test/Microsoft.Process.Environment.Tests/Microsoft.Process.Environment.Tests.vcxitems

```diff
@@ -0,0 +1,34 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Label="Globals">
+    <MSBuildAllProjects Condition="'$(MSBuildVersion)' == '' Or '$(MSBuildVersion)' &lt; '16.0'">$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+    <HasSharedItems>true</HasSharedItems>
+    <ItemsProjectGuid>{2a9dee02-db37-4b9c-a0da-b17541cb12c0}</ItemsProjectGuid>
+  </PropertyGroup>
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories);$(MSBuildThisFileDirectory)</AdditionalIncludeDirectories>
+    </ClCompile>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ProjectCapability Include="SourceItemsFromImports" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="$(MSBuildThisFileDirectory)EnvironmentManagerCentennialTests.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)EnvironmentManagerUWPTests.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)EnvironmentManagerWin32Tests.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClCompile Include="$(MSBuildThisFileDirectory)EnvironmentManagerCentennialTests.cpp" />
+    <ClCompile Include="$(MSBuildThisFileDirectory)EnvironmentManagerUWPTests.cpp" />
+    <ClCompile Include="$(MSBuildThisFileDirectory)EnvironmentManagerWin32Tests.cpp" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="$(MSBuildThisFileDirectory)PropertySheet.props" />
+  </ItemGroup>
+  <ItemGroup>
+    <Xml Include="$(MSBuildThisFileDirectory)AppxManifest.pkg.xml" />
+    <Xml Include="$(MSBuildThisFileDirectory)CentennialAppxManifest.pkg.xml" />
+    <Xml Include="$(MSBuildThisFileDirectory)CppTest.dll.manifest" />
```

Comment:
Why? Are you adding EnvironmentManager tests into CppTest? Please don't; make your own test project(s).

See previous comment

Reply:
I'll answer this question for posterity and to maybe figure out a better way to make a test class.

The reason why I added my tests to CppTest with a shared project is because that is what I saw AppLifecycleTest do.

Should I create my own dll for testing?

Reply:
Yes. Create your own project creating a DLL for your tests. Or multiple projects if necessary (DynamicDependencies has 7 test projects)

Add a feature subdir under test like under dev and put your test project in there. Or make subdirs under there if need be. I recommend the latter as it's easier if you later need to add a 2nd project e.g.

```
test\
    EnvironmentManager\
        EnvironmentManagerTest\
            EnvironmentManagerTest.vcxproj
            ...
```
or
```
test\
    EnvironmentManager\
        kitties\
            ...
        puppies\
            ...
```

Use your judgement. Main point is keep your tests in your own test project, separate from other features. TestExplorer makes it easy to run tests across features if/when needed but not drag down day-to-day inner-loop.